simple-proofs/verification.k

```diff
@@ -12,11 +12,9 @@ module VERIFICATION
 
   rule #Ceil({@L:KItem}:>List) => {isList(@L) #Equals true} #And #Ceil(@L) [anywhere, simplification]
 
-  rule #Ceil(#push(@E:Map, @X:UplcId, @I:Int)) =>
-       ({@X in_keys(@E) #Equals false} #Or {isList(@E[@X] orDefault .List) #Equals true})
-       #And #Ceil(@E) #And #Ceil(@X) #And #Ceil(@I) [simplification]
-
-  rule {#uplcHash(V1:Value) #Equals #uplcHash(V2:Value)} => {V1 #Equals V2} [simplification]
+  rule #Ceil(#push(@E:Map, @X:UplcId, @V:Value)) =>
+       ( { @X in_keys(@E) #Equals false } #Or ( { @X in_keys(@E) #Equals true } #And { isValue(@E[@X]) #Equals true } ) )
+       #And #Ceil(@E) #And #Ceil(@X) #And #Ceil(@V) [simplification]
```

Comment:
```suggestion
  rule #Ceil(#push(@E:Map, @X:UplcId, @V:Value)) =>
       #Ceil(@E) #And #Ceil(@X) #And #Ceil(@V) [simplification]
```
`#push` is a very simple map update now so all of the conditions should no longer be needed. It might be the case that this entire rule would no longer be needed.